analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -214,6 +170,59 @@ class Cargo(
             packageDir.startsWith(analysisRoot)
         } ?: false
 
+    private fun buildDependencyTree(
+        name: String,
+        version: String,
+        packages: Map<String, Package>,
+        metadata: JsonNode
+    ): PackageReference {
+        val node = metadata["packages"].single {
+            it["name"].textValue() == name && it["version"].textValue() == version
+        }
+        val dependencies = node["dependencies"]
+            .filter {
+                // Filter dev and build dependencies, because they are not transitive.
+                val kind = it["kind"].textValueOrEmpty()
+                kind != "dev" && kind != "build"
```

Comment:
Nit: Maybe invert the logic with `filterNot` ... `kind == "dev" || kind == "build"`

Reply:
Not applicable anymore after moving this to a helper function.